protostar/commands/test/test_environment_exceptions.py

```diff
@@ -29,6 +29,7 @@ class ReportedException(BaseException):
 
     def __init__(self, *args: object) -> None:
         self.metadata: List[ExceptionMetadata] = []
+        self.execution_info: Dict[str, Union[int, str]] = {}
```

Comment:
Why not reusing `metadata` field here? We could add metadata objects like `class FuzzRunsMetadata(ExceptionMetadata, ExecutionInfo)` and you could write `isinstance(..., ExecutionInfo)` in the loop in `FailingTestCase`.

BTW I'm starting to dislike the difference of handling succeeding and failing test and the duplication of logic which is a result of it...

Reply:
I created a new field because to print `fuzz_runs` next to `steps` and other execution properties etc. Statistics are easier to interpret when we know the number of observations.

> BTW I'm starting to dislike the difference of handling succeeding and failing test and the duplication of logic which is a result of it...

I don't like it, too. I created the task for that, and I plan to refactor after the release. https://github.com/software-mansion/protostar/issues/544

Reply:
> I created a new field because to print `fuzz_runs` next to `steps` and other execution properties etc. Statistics are easier to interpret when we know the number of observations.

Yeah, this part I got. The thing I am not fully liking is the fact that there is a new field `execution_info` added to the exception, instead of reusing the `metadata` field which is already existing.

Take into account that exception metadata must be pickleable and this property is probably easier to manage if all metadata-like things are stored in uniform way.

Reply:
Ok, I'll change to use `metadata` property and use `instanceof` in `FailedTestCase` to determine how to print metadata.

Reply:
Discussed privately. It will be refactored in https://github.com/software-mansion/protostar/issues/544.